ext/random/engine_mt19937.c

```diff
@@ -301,7 +301,7 @@ PHP_METHOD(Random_Engine_Mt19937, generate)
 
 	generated = engine->algo->generate(engine->status);
 	size = engine->status->last_generated_size;
-	if (engine->status->last_unsafe) {
+	if (EG(exception)) {
 		zend_throw_exception(spl_ce_RuntimeException, "Random number generation failed", 0);
 		RETURN_THROWS();
```

Comment:
tbh, I don't fully understand this bit, but it looks like an exception has happened elsewhere, and the code is replacing that exception with a new one. Is it possible to set the previous exception, so that the info isn't lost?

Reply:
> Is it possible to set the previous exception, so that the info isn't lost?

zend_throw_exception is smart enough to do that automatically:

https://github.com/php/php-src/blob/9090e2602eccc1edda02bfdb0eaf2b18b4a09e85/Zend/zend_exceptions.c#L182

This can be seen in action here:

https://github.com/php/php-src/blob/9090e2602eccc1edda02bfdb0eaf2b18b4a09e85/ext/random/tests/03_randomizer/user_unsafe.phpt#L135-L143

I plan to fully remove the `RuntimeException` here in a future PR, though, so that the underlying error of the broken engine is directly exposed to the user. This PR focuses on the cleanup work to make that possible, one piece at a time.

Reply:
Cool. TIL zend_throw_exception is smarter than me.